File: conformance/tests/arithmetic/arithmetic.runar.py
from runar import SmartContract, Bigint, public, assert_

class Arithmetic(SmartContract):
    target: Bigint

    def __init__(self, target: Bigint):
        super().__init__(target)
        self.target = target

    @public
    def verify(self, a: Bigint, b: Bigint):
        sum_: Bigint = a + b
        diff: Bigint = a - b
        prod: Bigint = a * b
        quot: Bigint = a // b
        result: Bigint = sum_ + diff + prod + quot
        assert_(result == self.target)

File: conformance/tests/arithmetic/arithmetic.runar.rb
require 'runar'

class Arithmetic < Runar::SmartContract
  prop :target, Bigint

  def initialize(target)
    super(target)
    @target = target
  end

  runar_public a: Bigint, b: Bigint
  def verify(a, b)
    sum = a + b
    diff = a - b
    prod = a * b
    quot = a / b
    result = sum + diff + prod + quot
    assert result == @target
  end
end
